### DISCUSSION TITLE
Autosave forms

### DISCUSSION THREAD

**Discussion by iainbeeston** [avo-hq/avo#1028] [Feature requests]
I'd really like an option to auto-save resources on the edit page in avo. I'd love to hear any advice on whether there is a workaround for this in avo (custom stimulus controllers?). If there is no obvious workaround, I'd be happy to contribute a PR to do it (but I'd also appreciate any suggestions on how the code should work)

**Comment by adrianthedev** [comment]
TBH this could be done with a custom Stimulus controller. I don't think you'd need to PR it even; only if you want to share. 

- Add a custom js file to the head from your asset pipeline. [Docs](https://docs.avohq.io/2.0/custom-asset-pipeline.html#sprockets)
- add a custom Stimulus controller. [Docs](https://docs.avohq.io/2.0/stimulus-integration.html#assign-stimulus-controllers-to-resource-views)

The controller flow would be like so:

- when the page loads start a timer
- When the timer executes it should:
- Disable all fields (you can target them through Stimulus targets). [Docs](https://docs.avohq.io/2.0/stimulus-integration.html#all-controllers-receive-the-view-value)
- submit the form
- If there are validation errors, they should be displayed
- If there aren't any errors, it will display a "success" message. 

There might be a few more things that come to mind:
- you have to take account for like prevent submitting when a user inputs data. 
- Maybe a countdown somehow (you can create a custom field). 
- Maybe a "disable auto-save" countdown

Let me know your thoughts. I'd loce to help you make this a reality. Maybe we'll push it in the package too. 
I'm open to making changes to Avo to accomodate this feature if needed.
